### PR TITLE
gnomeExtensions.system-monitor-next: Patch to fix runtime errors

### PR DESCRIPTION
**Pull request by andersk**
## Description of changes

This is required to get this extension to load successfully. It fixes the error “Requiring GTop, version none: Typelib file for namespace 'GTop' (any version) not found” and others.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/gnome/extensions/extensionOverridesPatches/system-monitor-next_at_paradoxxx.zero.gmail.com.patch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
++            let info = file.query_info(Gio.FILE_ATTRIBUTE_UNIX_IS_MOUNTPOINT,
++                Gio.FileQueryInfoFlags.NONE, null);
++            return info.get_attribute_boolean(Gio.FILE_ATTRIBUTE_UNIX_IS_MOUNTPOINT);
++        } catch (e) {
++            if (e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.NOT_FOUND)) {
++                return false;
++            }
++            throw e;
++        }
+     }
```
Comment (andersk) [inline pkgs/desktops/gnome/extensions/extensionOverridesPatches/system-monitor-next_at_paradoxxx.zero.gmail.com.patch:48]: Sent this part upstream:

- mgalgs/gnome-shell-system-monitor-applet#59

**Review by r-vdp — APPROVED** [review]
Nice, I tried it out and it's working well, thanks!

**Comment by piegamesde** [comment]
Change looks good, but it would be nice if you could also do the following things additionally:

- Last commit on `system-monitor` is over a year old, I think we can safely remove the manual packaging for now
- Add a rename/alias from `system-monitor` to `system-monitor-next`, as it is a pretty straightforward continuation despite the different slug. Actually, we might even think about renaming `system-monitor-next` into `system-monitor` completely because of this.
- Since you maintained `system-monitor` previously, it would be cool to add yourself as maintainer of `system-monitor-next` through the override

**Comment by andersk** [comment]
I’m not sure a rename/alias makes sense given that the upstream extension ID is different (`system-monitor@paradoxxx.zero.gmail.com` vs. `system-monitor-next@paradoxxx.zero.gmail.com`). Enabling one won’t enable the other, settings aren’t shared, etc.

**Comment by andersk** [comment]
Removed the manual packaging for the old system-monitor, and added myself as maintainer for system-monitor-next.

**Comment by AndrewKvalheim** [comment]
> settings aren’t shared

It does still use the [`org.gnome.shell.extensions.system-monitor`](https://github.com/mgalgs/gnome-shell-system-monitor-applet/blob/v3.9/system-monitor-next%40paradoxxx.zero.gmail.com/schemas/org.gnome.shell.extensions.system-monitor.gschema.xml) namespace.

**Pull request merged**